### ISSUE TITLE
TypeError: Cannot read property 'graphs' of undefined.

### ISSUE THREAD

**Issue by jjcad** [krispo/angular-nvd3#511]
This seems to be related to the code block around line 291 in angular-nvd3.js, where the comment is talking about doing something for compatibility with an older version of nvd3.

[plnkr example](http://plnkr.co/edit/Zjb0NNF4zuEd2xyx8iLH)

**Comment by jjcad** [comment]
Rolling back to older versions of d3, nvd3, and angular-nvd3 shows a different error, so maybe it's my code and not angular-nvd3?

`TypeError: nv.models[options.chart.type] is not a function`

[plnkr example 2](http://plnkr.co/edit/Fz9sDdALqSTTUQWhdKOl)

**Comment by shivaniviradia** [comment]
+1

**Comment by ghost** [comment]
+1
Reverted to version 1.0.8 and it's working fine now.

**Comment by crediblebytes** [comment]
I am having same issue. On website it says that 1.0.9 is compatible with nvd3 1.8.1+ but it is not. I get the error described above. Even when I use exactly 1.8.1 version of nvd3 it does not work. However rolling back to 1.0.8 as suggested fixes the problem.
- angular-nvd3 v1.0.9 + nvd3 v1.8.4 + d3 v4.2.6 (BROKEN)
- angular-nvd3 v1.0.9 + **nvd3 v1.8.1** + d3 v4.2.6 (BROKEN)
- **angular-nvd3 v1.0.8** + nvd3 v1.8.4 + d3 v4.2.6 (WORKS)

**Comment by krispo** [comment]
@jjcad in your demo you include d3.js script in wrong way:

```
<script> src="https://cdnjs.cloudflare.com/ajax/libs/d3/3.5.3/d3.js"</script>
```

fix to

```
<script src="https://cdnjs.cloudflare.com/ajax/libs/d3/3.5.3/d3.js"></script>
```

and everything works fine.

**Comment by jjcad** [comment]
That's incredibly embarassing. I can't believe I didn't notice that.

**Comment by crediblebytes** [comment]
This still seems to have problems. I get errors when trying to use the latest versions as described above.
http://plnkr.co/edit/Ygt5n30WGuaIaw9IY56o?p=preview

**Comment by marcosicp** [comment]
any news with this error? @krispo

**Comment by nstrelow** [comment]
also having that problem

**Comment by charbs-io** [comment]
Still having the same issue. Downgrading to 1.0.8 as per @goodies4uall 's suggestion fixes it.

**Comment by ximemodelit** [comment]
Yep, I'm still seeing the error.

**Comment by levydev** [comment]
New comer.    angular_nvd3 tutorial does not work

**Comment by janey0536** [comment]
same with levydev. Tried the 'basic usage' and it didn't work.

**Comment by tomergev** [comment]
+1

**Comment by tkserver** [comment]
Just started using angular-nvd3.  Ran into this issue and found this thread via google.  I too solved by changing to 1.0.8, for what it's worth.

**Comment by harisanker1780** [comment]
It is working after changing to 1.0.8
